onmt/translate/translator.py

```diff
@@ -345,9 +386,16 @@ def unbottle(m):
             inp = inp.unsqueeze(2)
 
             # Run one step.
-            dec_out, dec_states, attn = self.model.decoder(
-                inp, memory_bank, dec_states, memory_lengths=memory_lengths)
+            if self.self_attn_type == "average":
```

Comment:
Would prefer this not to be in `translate.py` which is a key pathway. Any way you can have this work automatically?